.github/workflows/kafka_demo_test.yml

```diff
@@ -41,34 +41,55 @@ jobs:
       key-release-image: ${{ steps.extract-envs.outputs.key-release-image }}
       consumer-image: ${{ steps.extract-envs.outputs.consumer-image }}
       producer-image: ${{ steps.extract-envs.outputs.producer-image }}
+      workflow-id: ${{ steps.generate-id.outputs.id }}
     steps:
       - name: Checkout
         uses: actions/checkout@v4
 
+      - name: Get Workflow ID
+        id: generate-id
+        run: echo "id=$(openssl rand -hex 8 | tr -d '\n')" >> $GITHUB_OUTPUT
+
       - name: Extract Envs 
         id: extract-envs
         run: | 
           echo "Extracting Environment Variables"
           echo "Triggering event is ${{ github.event_name }}"
           if [[ "${{ github.event_name }}" == "pull_request" ]]; then
             echo "key-release-image=mcr.microsoft.com/aci/skr:2.7" >> $GITHUB_OUTPUT
-            echo "consumer-image=mcr.microsoft.com/acc/samples/kafka/consumer:1.0" >> $GITHUB_OUTPUT
-            echo "producer-image=mcr.microsoft.com/acc/samples/kafka/producer:1.0" >> $GITHUB_OUTPUT
+            echo "consumer-image=${{ secrets.AZURE_REGISTRY_URL }}/private/acc/samples/kafka/consumer:${{ steps.generate-id.outputs.id }}" >> $GITHUB_OUTPUT
+            echo "producer-image=${{ secrets.AZURE_REGISTRY_URL }}/private/acc/samples/kafka/producer:${{ steps.generate-id.outputs.id }}" >> $GITHUB_OUTPUT
           else
             echo "key-release-image=${{ format('{0}/{1}', inputs.sidecar-registry, inputs.key-release-image) }}" >> $GITHUB_OUTPUT
             echo "consumer-image=${{ inputs.consumer-image }}" >> $GITHUB_OUTPUT
             echo "producer-image=${{ inputs.producer-image }}" >> $GITHUB_OUTPUT
           fi
 
+  push_images:
+    name: Push Images to Private Registry for Testing
+    needs: [prepare]
+    uses: ./.github/workflows/_push_image.yml
+    secrets: inherit
+    strategy:
+      matrix:
+        image: ["kafka/consumer", "kafka/producer"]
+    with:
+      image_tag: ${{ needs.prepare.outputs.workflow-id }}
+      image: ${{ matrix.image }}
+      repo_type: private
+      docker_context: ${{ matrix.image }}
+
   create-aks-cluster:
     name: Create AKS Cluster
-    needs: [prepare]
+    needs: [prepare, push_images]
```

Comment:
Does this mean that we always have to build and push images before running the test? We should configure the pipeline in such a way that building and pushing images are optional. We can indicate image tags to use and skips the build and push image step when image tags exist

Reply:
Good point, I can have the push images step only run for PRs and that should do it